frontend/public/components/instantiate-template.tsx

```diff
@@ -241,136 +220,130 @@ class TemplateForm_ extends React.Component<
       name: secret.metadata.name,
       ns: secret.metadata.namespace,
     });
-  }
+  };
 
-  save = (event: React.FormEvent<EventTarget>) => {
+  const save = (event: React.FormEvent<EventTarget>) => {
     event.preventDefault();
-    const { namespace } = this.state;
     if (!namespace) {
-      this.setState({ error: 'Please complete all fields.' });
+      setError('Please complete all fields.');
       return;
     }
-    const { activePerspective, perspectiveExtensions } = this.props;
 
-    this.setState({ error: '', inProgress: true });
-    this.createTemplateSecret()
+    setError('');
+    setInProgress(true);
+
+    createTemplateSecret()
       .then((secret: K8sResourceKind) => {
-        return this.createTemplateInstance(secret).then(
-          async (templateInstance: K8sResourceKind) => {
-            await this.updatesecretOwnerRef(secret, templateInstance);
-            this.setState({ inProgress: false });
-            const activeExtension = perspectiveExtensions.find(
-              (p) => p.properties.id === activePerspective,
-            );
-            const url = (await activeExtension.properties.importRedirectURL())(namespace);
-            history.push(url);
-          },
-        );
+        return createTemplateInstance(secret).then(async (templateInstance: K8sResourceKind) => {
+          await updatesecretOwnerRef(secret, templateInstance);
+          setInProgress(false);
+          const activeExtension = perspectiveExtensions.find(
+            (p) => p.properties.id === activePerspective,
+          );
+          const url = (await activeExtension.properties.importRedirectURL())(namespace);
+          history.push(url);
+        });
       })
-      .catch((err) => this.setState({ inProgress: false, error: err.message }));
+      .catch((err) => {
+        setInProgress(false);
+        setError(err.message);
+      });
   };
 
-  render() {
-    const { obj, t } = this.props;
-    if (obj.loadError) {
-      return (
-        <LoadError
-          message={obj.loadError.message}
-          label={t('public~Template')}
-          className="loading-box loading-box__errored"
-        />
-      );
-    }
+  if (obj.loadError) {
+    return (
+      <LoadError
+        message={obj.loadError.message}
+        label={t('public~Template')}
+        className="loading-box loading-box__errored"
+      />
+    );
+  }
 
-    if (!obj.loaded) {
-      return <LoadingBox />;
-    }
+  if (!obj.loaded) {
+    return <LoadingBox />;
+  }
 
-    const template: TemplateKind = obj.data;
-    const parameters = template.parameters || [];
+  const template: TemplateKind = obj.data;
+  const params = template.parameters || [];
```

Comment:
Assuming the simplification mentioned above works, the above variable could be kept as `parameters` as in original code.

Reply:
Keeping this variable as params because parameters is a separate variable of a different type.